Sources/SourceKitLSP/Documentation/DocumentationManager.swift

```diff
@@ -0,0 +1,274 @@
+    } else if let child = DocumentableSymbolFinder.find(
+      in: memberBlock.children(viewMode: .sourceAccurate),
+      at: cursorPosition
+    ) {
+      setResult(result: child)
+    } else if node.range.contains(cursorPosition) {
+      setResult(node: node, position: name.positionAfterSkippingLeadingTrivia)
+    }
+    return .skipChildren
+  }
+
+  override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+    visitNamedDeclWithMemberBlock(node: node, name: node.name, memberBlock: node.memberBlock)
+  }
+
+  override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+    visitNamedDeclWithMemberBlock(node: node, name: node.name, memberBlock: node.memberBlock)
+  }
+
+  override func visit(_ node: ActorDeclSyntax) -> SyntaxVisitorContinueKind {
+    visitNamedDeclWithMemberBlock(node: node, name: node.name, memberBlock: node.memberBlock)
+  }
+
+  override func visit(_ node: EnumDeclSyntax) -> SyntaxVisitorContinueKind {
+    visitNamedDeclWithMemberBlock(node: node, name: node.name, memberBlock: node.memberBlock)
+  }
+
+  override func visit(_ node: ProtocolDeclSyntax) -> SyntaxVisitorContinueKind {
+    visitNamedDeclWithMemberBlock(node: node, name: node.name, memberBlock: node.memberBlock)
+  }
+
+  override func visit(_ node: FunctionDeclSyntax) -> SyntaxVisitorContinueKind {
+    let symbolPosition = node.name.positionAfterSkippingLeadingTrivia
+    if node.range.contains(cursorPosition) || cursorPosition < symbolPosition {
+      setResult(node: node, position: symbolPosition)
+    }
+    return .skipChildren
+  }
+
+  override func visit(_ node: MemberBlockSyntax) -> SyntaxVisitorContinueKind {
+    let range = node.leftBrace.endPositionBeforeTrailingTrivia..<node.rightBrace.positionAfterSkippingLeadingTrivia
+    guard range.contains(cursorPosition) else {
+      return .skipChildren
+    }
+    return .visitChildren
+  }
+
+  override func visit(_ node: InitializerDeclSyntax) -> SyntaxVisitorContinueKind {
+    let symbolPosition = node.initKeyword.positionAfterSkippingLeadingTrivia
+    if node.range.contains(cursorPosition) || cursorPosition < symbolPosition {
+      setResult(node: node, position: symbolPosition)
+    }
+    return .skipChildren
+  }
+
+  override func visit(_ node: EnumCaseDeclSyntax) -> SyntaxVisitorContinueKind {
+    for element in node.elements {
+      let symbolPosition = element.name.positionAfterSkippingLeadingTrivia
+      if element.range.contains(cursorPosition) || cursorPosition < symbolPosition {
+        setResult(node: node, position: symbolPosition)
+      }
+    }
+    return .skipChildren
+  }
+
+  override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+    // A variable declaration is only documentable if there is only one pattern binding
+    guard node.bindings.count == 1,
+      let identifier = node.bindings.first!.pattern.as(IdentifierPatternSyntax.self)
+    else {
```

Comment:
You can simplify this slightly

```suggestion
    guard let identifier = node.bindings.only.pattern.as(IdentifierPatternSyntax.self) else {
```

Reply:
Done.